mautic/mautic
# [Open Source Fridays] Accessibility enhancement for the top navbar while using screen readers

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR enhances user experience for people with screen readers. Specifically:
1. Avoid Mautic logo being selected and read as “graphic”
2. Refined the code to ensure that list items (`<li>`) within the navigation bar do not interrupt the flow of the screen reader, allowing for seamless transition to subsequent links.
3. Adjusted icon elements to be bypassed by screen readers when they do not convey essential information, reducing unnecessary stops and enhancing user experience.
4. Enhanced the notification system with a non-visible label that screen readers can announce, alerting users to new notifications effectively.
5. Ensured that screen readers communicate the presence of position located beneath the user’s name, providing context and clarity.
6. Made certain that users are informed by the screen reader of the existence of a profile picture, personalizing their experience.
7. Clarified that the user’s name functions as an interactive link to their account menu, making navigation more intuitive.
8. Established a clear distinction between links that navigate to a different page and buttons that trigger actions on the current page, aiding in user expectation management.
9. Improved the dashboard widgets dropdown to be screen reader-friendly, ensuring that it is properly associated with its corresponding menu and actions when activated.
10. Removes maximum scale property from viewport to ensure accessibility tools can use zooming features without issues 

Pending tasks:

- [ ] Solve the weird behaviour when screen readers try to read the hidden search input
- [ ] Add aria-label to page actions dropdown, which needs a dev to rewrite the function so we can put a translation string there



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Activate your screen reader (Windows + Ctrl + Enter on Windows 11)
3. Navigate to the homepage.
4. Use the screen reader’s navigation commands to reach the page elements.
5. Confirm that the screen reader skip the logo correctly and does not read it as a graphic.
6. Focus on the navigation bar.
7. Use the arrow keys to move through the items.
8. Confirm the screen reader does not stop prematurely and reads all relevant links without interruption.
9. Navigate where icons are present.
11. Observe if the screen reader skips non-essential icons.
12. Confirm that only informative icons are read out.
13. Trigger a new notification (e.g., by receiving a message or alert).
14. Navigate to the asterisk icon to listen for the screen reader to announce there is a new notification.
15. Locate the area beneath the user’s name where the position is displayed.
16. Confirm that the screen reader announces the position as part of the user details.
17. Find the profile picture element.
18. Check that the screen reader informs about the presence of a profile picture.
19. Select the user’s link.
20. Confirm that the screen reader indicates that it has pop up and leads to the account menu.
21. Interact with all the previous elements by navigating to them.
22. Note the different responses from the screen reader for each type of element.
23. Navigate to the dashboard widgets dropdown.
24. Activate the dropdown.
25. Confirm the screen reader reads the associated menu and its options when clicked.
